<a href="https://colab.research.google.com/github/mahunz/MDBDS_M2/blob/main/M2_AI2_HunzikerMartin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1> Estadistica Descriptiva </h1>
<h3>M2_AI3_Coronavirus</h3>
<h3>Alumno: Martín Hunziker</h3>




### Librerias

In [ ]:
import pandas as pd
import numpy as np
from functools import reduce
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns; sns.set()  # para el estilo de graficos
import math
import scipy.stats as ss

sns.set_theme(style="darkgrid")

### Lectura de los datos y Análisis de las variables del dataset

Leemos el archivo precio_gasolina.csv desde github y visualizamos la cabezera

In [26]:
# Carga del fichero desde el enlace web y creación del dataframe
url = 'https://raw.githubusercontent.com/md-lorente/data/master/worlddometer_covid_ue.csv'

# Creación del Dataframe
DS_Covid = pd.read_csv(url,sep = ';', decimal = ',')

# Visualización del dataframe (la cabecera)
DS_Covid.head()

Num Country Total Cases  New Cases  Total Deaths  New Deaths  \
0    0  Europe   3,543,710     10.493       207.347       149.0   
1    1  Russia     990,326      4.980        17.093        68.0   
2    2   Spain     455,621        NaN        29.011         NaN   
3    3      UK     332,752        NaN        41.498         NaN   
4    4  France     272,530        NaN        30.602         NaN   

  Total Recovered Active Cases  Serious Critical  Total Cases per 1M pop  \
0       2,063,335    1,273,028             5.619                     NaN   
1         806,982      166,251             2.300                   6.786   
2             NaN          NaN           751.000                   9.744   
3             NaN          NaN            60.000                   4.897   
4          86,177      155,751           387.000                   4.174   

   Deaths per 1M pop Total Tests Tests per 1M pop  Population   
0                NaN         NaN              NaN          NaN  
1              117.0  36,400,000          249,409  145,944,843  
2              620.0   8,517,446          182,161   46,757,783  
3              611.0  16,273,209          239,508   67,944,393  
4              469.0   6,175,223           94,571   65,297,356

In [27]:
# Resumen información del dataframe
print(DS_Covid.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49 entries, 0 to 48
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Num                     49 non-null     int64  
 1   Country                 49 non-null     object 
 2   Total Cases             49 non-null     object 
 3   New Cases               17 non-null     float64
 4   Total Deaths            46 non-null     float64
 5   New Deaths              7 non-null      float64
 6   Total Recovered         45 non-null     object 
 7   Active Cases            45 non-null     object 
 8   Serious Critical        37 non-null     float64
 9   Total Cases per 1M pop  48 non-null     float64
 10  Deaths per 1M pop       45 non-null     float64
 11  Total Tests             47 non-null     object 
 12  Tests per 1M pop        47 non-null     object 
 13  Population              48 non-null     object 
dtypes: float64(6), int64(1), object(7)
memory us

In [28]:
# Contando los valores  NaN en todas las columnas
nan_count = DS_Covid.isna().sum()

print(nan_count)

Num                        0
Country                    0
Total Cases                0
New Cases                 32
Total Deaths               3
New Deaths                42
Total Recovered            4
Active Cases               4
Serious Critical          12
Total Cases per 1M pop     1
Deaths per 1M pop          4
Total Tests                2
Tests per 1M pop           2
Population                 1
dtype: int64


En el dataset no hay valores NA que puedan impedir el cálculo de los estadísticos

In [29]:
# Analizamos los registros únicos de las columnas
DS_Covid.apply(lambda col: col.unique())

Num                       [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,...
Country                   [Europe, Russia, Spain, UK, France, Italy, Ger...
Total Cases               [3,543,710, 990,326, 455,621, 332,752, 272,530...
New Cases                 [10.493, 4.98, nan, 2.096, 952.0, 647.0, 631.0...
Total Deaths              [207.347, 17.093, 29.011, 41.498, 30.602, 35.4...
New Deaths                         [149.0, 68.0, nan, 35.0, 39.0, 5.0, 1.0]
Total Recovered           [2,063,335, 806,982, nan, 86,177, 208,224, 217...
Active Cases              [1,273,028, 166,251, nan, 155,751, 23,156, 15,...
Serious Critical          [5.619, 2.3, 751.0, 60.0, 387.0, 79.0, 245.0, ...
Total Cases per 1M pop    [nan, 6.786, 9.744, 4.897, 4.174, 4.415, 2.897...
Deaths per 1M pop         [nan, 117.0, 620.0, 611.0, 469.0, 587.0, 112.0...
Total Tests               [nan, 36,400,000, 8,517,446, 16,273,209, 6,175...
Tests per 1M pop          [nan, 249,409, 182,161, 239,508, 94,571, 140,7...
Population  

In [31]:
DS_Covid.columns


Index(['Num', 'Country', 'Total Cases', 'New Cases', 'Total Deaths',
       'New Deaths', 'Total Recovered', 'Active Cases', 'Serious Critical',
       'Total Cases per 1M pop', 'Deaths per 1M pop', 'Total Tests',
       'Tests per 1M pop', 'Population '],
      dtype='object')

In [30]:
# Analizamos los datos
DS_Covid.describe()

Num   New Cases  Total Deaths  New Deaths  Serious Critical  \
count  49.00000   17.000000     46.000000    7.000000         37.000000   
mean   24.00000  202.033471    196.501326   42.571429         89.916730   
std    14.28869  285.670090    265.986505   53.090668        175.918556   
min     0.00000    2.096000      1.000000    1.000000          1.000000   
25%    12.00000   10.493000      9.918250    3.000000          3.000000   
50%    24.00000   34.000000     45.000000   35.000000         19.000000   
75%    36.00000  292.000000    272.250000   53.500000         61.000000   
max    48.00000  952.000000    990.000000  149.000000        751.000000   

       Total Cases per 1M pop  Deaths per 1M pop  
count               48.000000          45.000000  
mean                68.105542         194.605267  
std                212.860839         214.483793  
min                  1.058000           1.237000  
25%                  2.896500          48.000000  
50%                  4.682500          96.000000  
75%                  8.330250         276.000000  
max                958.000000         853.000000

El tipo de variables del dataset es:

1. **Num:** es codigo del dataset. El código es único, consecutivo y entero. Es una **variable cuantitativa de tipo discreto.**

2. **Country:** Es una variable clasificada con palabras que no sigue ningún orden en específico. Es una variable **cualitativa nominal**.

3. **Total Cases** Es el cantidad de casos total de COVID. En un número entero. **variable cuantitativa de tipo discreto.**

4. **New Cases** Es la cantidad de nuevos casos de COVID. En un número entero. **variable cuantitativa de tipo discreto.**

5. **Total Deaths** Es la cantidad total de muertes por COVID. En un número entero. **variable cuantitativa de tipo discreto.**

6. **New Deaths** Es la cantidad de nuevas muertes por COVID. En un número entero. **variable cuantitativa de tipo discreto.**

7. **Total Recovered** Es la cantidad de casos recuperados de COVID. En un número entero. **variable cuantitativa de tipo discreto.**

8. **Active Cases** Es la cantidad de casos Activos de COVID. En un número entero. **variable cuantitativa de tipo discreto.**

9. **Serious Critical** Es la cantidad de casos en estado críticos de COVID. En un número entero. **variable cuantitativa de tipo discreto.**

10. **Total Cases per 1M pop** Es la cantidad total de casos por millon de habitantes. Es una variable **cuantitativa continua.**  Nótese: las variables tiene decimales.

11. **Deaths per 1M pop** Es la cantidad total de muertos por millon de habitantes. Es una variable **cuantitativa continua.**  Nótese: las variables tiene decimales.

12. **Total Tests** Es la cantidad total de test realizados. En un número entero. **variable cuantitativa de tipo discreto.**

13. **Tests per 1M pop** Es la cantidad de test por millon de habitantes. Es una variable **cuantitativa continua.**  Nótese: las variables tiene decimales.

14. **Poblacion** Es la población del país. En un número entero. **variable cuantitativa de tipo discreto.**


## Analizar

* La probabilidad de sobrevivir en cualquiera de los seis países europeos seleccionados para el estudio. Se debe aportar su fórmula y su sustitución. Analizar los resultados obtenidos (15%)
* Si seleccionamos una de las personas fallecidas, ¿cuál es la probabilidad de que residiera en España? ¿Y en el resto de países? Se debe aportar su fórmula y su sustitución. (70%)
* Aportar el árbol de decisión, añadir una explicación de cómo funciona, el resultado y un comentario o análisis del resultado obtenido. (15%)
